pypsa/networkclustering.py

```diff
@@ -172,7 +180,14 @@ def aggregatelinegroup(l):
         v_nom_s = buses.loc[list(l.name),'v_nom'].max()
 
         voltage_factor = (np.asarray(network.buses.loc[l.bus0,'v_nom'])/v_nom_s)**2
-        length_factor = (length_s/l['length'])
+        
+        # Could be lines (not transformer) that is 0 distance before and after clustering, but also with non-0 original r, x, g, b values, due to GIS resolution or other reasons.
+        if np.asarray(l['length']).min() == 0 and np.asarray(l['x']).prod() != 0:
+            length_factor = 1
+        else:
+            length_factor = (length_s/l['length'])
+        
+        # length_factor = (length_s/l['length'])
```

Comment:
I am still wondering about this one. Shouldn't we make sure that only where the length is 0, the length_factor is set to 1? Like the following
```suggestion
    length_factor = (length_s/l['length']).where(l['length'] != 0, 1)
```

Reply:
Much more elegant! I was trying to rule out the possibility that there are branches with both 0 length and 0 parameters in original data. In this case, shall we somehow report out the error? That was my original thought. Agree that it is not the best way to let it give warnings on "dividing 0" to give error messages, maybe you would consider add a logic for data sanity check here?

Also, does l['length'] != 0 means every element in l['length'] should not be zero? I am not exactly sure but if so then all good. Just trying to not miss any 0 entries.

Reply:
You are right about the divide by zero warning! Let me make a better suggestion.

Indeed the l['length'] != 0 returns a series where every element is checked. So with the suggested change, we fill up ones everywhere where the length of a line is zero.